operator/pkg/securedcluster/values/translation/translation.go

```diff
@@ -275,8 +275,6 @@ func (t Translator) getCollectorContainerValues(collectorContainerSpec *platform
 		switch *c {
 		case platform.CollectionEBPF:
 			cv.SetStringValue("collectionMethod", storage.CollectionMethod_EBPF.String())
-		case platform.CollectionKernelModule:
```

Comment:
I have a concern about backward capability in the operator.
Maybe we can keep `CollectionKernelModule` and mark it as deprecated in the comment. Also map `CollectionKernelModule` to `EBPF`. Then delete it in 4.x version.
Not sure about such approach though

Reply:
@kurlov - I've implemented a version of your idea where the collection method is still removed but we still check for the legacy value and switch to eBPF in this case. I've tested it locally and it seems to be working as expected, and I'm interested to hear what you think of my implementation